modin/backends/pandas/query_compiler.py

```diff
@@ -388,16 +388,55 @@ def merge(self, right, **kwargs):
         -----
         See pd.merge or pd.DataFrame.merge for more info on kwargs.
         """
-        right = right.to_pandas()
+        how = kwargs.get("how", "inner")
+        on = kwargs.get("on", None)
+        left_on = kwargs.get("left_on", None)
+        right_on = kwargs.get("right_on", None)
+        left_index = kwargs.get("left_index", False)
+        right_index = kwargs.get("right_index", False)
+        sort = kwargs.get("sort", False)
 
-        sort = kwargs.get("sort")
-        kwargs["sort"] = not sort if sort else sort
+        if how in ["left", "inner"] and left_index is False and right_index is False:
+            right = right.to_pandas()
 
-        def map_func(left, right=right, kwargs=kwargs):
-            return pandas.merge(left, right, **kwargs)
+            kwargs["sort"] = not sort if sort else sort
```

Comment:
`not sort if sort else sort` is always `False`. Is it what is meant here?

Reply:
Yes, we don't want to sort cells within execution of function on partitions in order to avoid additional overhead. The logic regarding sorting is below.

Reply:
Then should this line be replaced with `kwargs["sort"] = False`?

Reply:
Yes, it makes sense, done. Thanks!